services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java

```diff
@@ -362,25 +366,30 @@ public Response export(UriInfo uriInfo, RealmModel realm, String format) {
 
             List<Element> signingKeys = new LinkedList<>();
             List<Element> encryptionKeys = new LinkedList<>();
-
             session.keys().getKeysStream(realm, KeyUse.SIG, Algorithm.RS256)
                     .filter(Objects::nonNull)
                     .filter(key -> key.getCertificate() != null)
                     .sorted(SamlService::compareKeys)
-                    .forEach(key -> {
+                    .map(key -> {
                         try {
-                            Element element = SPMetadataDescriptor
-                                    .buildKeyInfoElement(key.getKid(), PemUtils.encodeCertificate(key.getCertificate()));
-                            signingKeys.add(element);
-
-                            if (key.getStatus() == KeyStatus.ACTIVE) {
-                                encryptionKeys.add(element);
-                            }
+                            return SPMetadataDescriptor.buildKeyInfoElement(key.getKid(), PemUtils.encodeCertificate(key.getCertificate()));
+                        } catch (ParserConfigurationException e) {
+                            logger.warn("Failed to export SAML SP Metadata!", e);
+                            throw new RuntimeException(e);
+                        }
+                    }).forEach(signingKeys::add);
+            session.keys().getKeysStream(realm, KeyUse.ENC, JWEConstants.RSA_OAEP)
```

Comment:
A bit odd to use `JWEConstants` here. Perhaps we should add `RSA-OAEP` to `Algorithm` instead?